src/rtk/features/singleRights/singleRightsSlice.ts

```diff
@@ -34,31 +34,30 @@ interface detailParams {
   name: string;
   value: string;
 }
-
-interface chosenService {
+export interface ChosenService {
   accessCheckResponses?: delegationAccessCheckResponse[];
   service?: ServiceResource;
-  status?: 'Delegable' | 'NotDelegable';
+  status?: 'Delegable' | 'NotDelegable' | 'PartiallyDelegable';
   errorCode?: string;
 }
 
-interface chosenServices {
-  chosenServices: chosenService[];
+interface chosenServiceList {
+  chosenServices: ChosenService[];
```

Comment:
Also, considering this includes nondelegable objects that are not and cannot actually be chosen, maybe this too should be called something else?
Maybe servicesWithKnownStatus? Not exactly ideal, but at least it gets the point across 😅

Reply:
Or we could have two lists, one with notDelegable and one for chosenServices

Reply:
Don't think it's a good idea to have two lists as we need to maintain both those lists. I don't think servicesWithKnownStatus is a better wording. It should be a general word that's for both non delegable and delegable services that is chosen, and that word is ChosenServices right?  If that's confusing it shouldn't be confusing when looking at the ChosenService interface

Reply:
Why would you expect all chosenServices to show up in ChooseRightsPage when the ones with status NonDelegable explicitly are filtered out in the code on that page? 😅

Reply:
My logic is that, if the list includes services that aren't chosen (and NonDelegable services cannot be chosen), then the list is per definition not a list of chosen services, and thus, shouldn't be named like that.
It doesn't really have anything to do with the fact that the NonDelegable services are filtered out in the rights page. I just think the name is misleading for what it contains.
Either it contains only chosen services - that is, services that are to be delegated - or it contains more than that, in which case I think the name is misleading. I would expect the chosen services to be the ones added into the 'shopping cart' only.

It's not a major deal though, so feel free to mark this conversation as resolved, I just think it can get tiresome to have to check the interface every time you read the code where this state is used or if you're debugging and looking at the redux state in the browser.

Reply:
Ahh yeah now I see what you mean, what about ServicesWithStatus?